src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/ExplicitInitializationCheck.java

```diff
@@ -90,9 +90,7 @@ public void visitToken(DetailAST ast) {
             final DetailAST assign = ast.findFirstToken(TokenTypes.ASSIGN);
             final DetailAST exprStart =
                 assign.getFirstChild().getFirstChild();
-            final DetailAST type = ast.findFirstToken(TokenTypes.TYPE);
-            if (isObjectType(type)
-                && exprStart.getType() == TokenTypes.LITERAL_NULL) {
+            if (exprStart.getType() == TokenTypes.LITERAL_NULL) {
```

Comment:
Removing this code should be valid.

We can't assign `null` to non-objects, so the check if it is an object is redundant, and `null` assignment should be enough.

Reply:
confirmed.